### ISSUE TITLE
False Negative: Overflow cases that compute in `int` before widening to `long` are not caught.

### ISSUE THREAD

**Issue by Carlson-JLQ** [google/error-prone#5637]
# False Negative: Overflow cases that compute in `int` before widening to `long` are not caught.

Version
2.48.0

## Checker
- Checker id: `IntLongMath`
- Checker description: Detects when an int-typed arithmetic expression may overflow before being assigned to a long-typed variable, field, or return value.

## Description of the false negative
`diff/output` marks 1 false negatives for `IntLongMath`. Across these samples, the benchmark is still exercising the same underlying bug pattern, but the implementation appears to lose track of it once the source deviates from one canonical shape. The misses read more like matching gaps than deliberate exclusions.

### `PosCase1.java`
Source: `output/codeql/IntMultToLong/src/main/java/scensct/core/pos/PosCase1.java`

The sample keeps the benchmark's target pattern but expresses it in a slightly non-canonical source form. The risky pattern is still present in the code, so the absence of a report points to a matching gap rather than a benign variant. That is also how the benchmark classifies the sample: Multiplication of two int constants whose product definitely exceeds Integer.MAX_VALUE, then implicitly converted to long should be flagged as overflow.

```java
// Multiplication of two int constants whose product definitely exceeds Integer.MAX_VALUE, then implicitly converted to long should be flagged as overflow.
package scensct.core.pos;

public class PosCase1 {
    public static void main(String[] args) {
        // Both operands are int constants: 1000000 * 3000000 > Integer.MAX_VALUE
        long result = 1000000 * 3000000; // Implicit conversion to long after overflow // [REPORTED LINE]
        System.out.println(result);
    }
}
```

## Cause analysis
The rule is documented as follows: Detects when an int-typed arithmetic expression may overflow before being assigned to a long-typed variable, field, or return value.

 The missed cases suggest that the implementation is narrower than the rule description and too dependent on one preferred AST shape.

Once the same behavior is expressed through a helper, an overload, a modifier such as `abstract` or `native`, a different comment position, or a slightly rearranged control-flow structure, the report disappears even though the benchmark still treats the case as in-scope.

For `IntLongMath`, the practical improvement would be to generalize the match so it follows the underlying behavior rather than one exact spelling of it.

## References
None known.
